pulsar-client/src/main/java/org/apache/pulsar/client/impl/BatchMessageContainerImpl.java

```diff
@@ -324,10 +324,14 @@ public OpSendMsg createOpSendMsg() throws IOException {
     protected void updateAndReserveBatchAllocatedSize(int updatedSizeBytes) {
         int delta = updatedSizeBytes - batchAllocatedSizeBytes;
         batchAllocatedSizeBytes = updatedSizeBytes;
-        if (delta != 0) {
+        if (delta > 0) {
```

Comment:
Maybe this will look more graceful

```
        if (producer != null) {
            if (delta > 0){
                producer.client.getMemoryLimitController().forceReserveMemory(delta);
            } else if (delta < 0) {
                producer.client.getMemoryLimitController().releaseMemory(-delta);
            }
        }
```

Reply:
I have addressed your comment. PTAL. Thanks.